File: packages/dashbuilder/appformer/uberfire-client-api/src/main/java/org/uberfire/client/util/Clipboard.java
/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.uberfire.client.util;

import javax.enterprise.context.Dependent;

import elemental2.dom.HTMLInputElement;
import elemental2.dom.HTMLTextAreaElement;

@Dependent
public class Clipboard {

    public boolean copy(final HTMLInputElement element) {
        element.select();
        return copy();
    }

    public boolean copy(final HTMLTextAreaElement element) {
        element.select();
        return copy();
    }

    public native boolean copy() /*-{
        return $doc.execCommand("Copy");
    }-*/;
}

File: packages/boxed-expression-component/tests/e2e/__fixtures__/clipboard.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { BrowserContext, Page } from "@playwright/test";

export class Clipboard {
  constructor(public page: Page) {}

  public async copy() {
    const modifier = process.platform === "darwin" ? "Meta" : "Control";
    await this.page.keyboard.press(`${modifier}+KeyC`);
  }

  public async cut() {
    const modifier = process.platform === "darwin" ? "Meta" : "Control";
    await this.page.keyboard.press(`${modifier}+KeyX`);
  }

  public async paste() {
    const modifier = process.platform === "darwin" ? "Meta" : "Control";
    await this.page.keyboard.press(`${modifier}+KeyV`);
  }

  public async setup(context: BrowserContext, browserName: string) {
    if (browserName === "chromium") {
      await context.grantPermissions(["clipboard-read", "clipboard-write"]);
    }
  }

  public use() {}
}
